### PR TITLE
[RFC] Improve the UrlGeneratorTest

### PR DESCRIPTION
**Pull request by aschempp**

### CODE REVIEW

File: src/Routing/UrlGenerator.php [hunk 1]
```diff
@@ -141,6 +141,8 @@ private function prepareAlias($alias, array &$parameters)
         $hasAutoItem = false;
         $autoItems = $this->getAutoItems($parameters);
 
+        unset($parameters['auto_item']);
```
Comment (leofeyer) [inline src/Routing/UrlGenerator.php]: Why did you change this? It has worked without it, hasn't it?

File: tests/Routing/UrlGeneratorTest.php [hunk 2]
```diff
@@ -364,4 +356,32 @@ private function getGenerator($prependLocale = false, $returnArgument = 1, $useA
...
     }
+
+    /**
+     * @param array $expectedParameters
+     * @param       $expectedRoute
+     * @param int   $referenceType
+     *
+     * @return \PHPUnit_Framework_MockObject_MockObject|UrlGeneratorInterface
+     */
+    private function expectRouter(
```
Comment (leofeyer) [inline tests/Routing/UrlGeneratorTest.php]: CS: no line breaks in method definitions. 😄

File: src/Routing/UrlGenerator.php (hunk 1 above)
Comment (aschempp) [reply src/Routing/UrlGenerator.php]: No, the test failed after correctly implementing it 😉

File: tests/Routing/UrlGeneratorTest.php (hunk 2 above)
Comment (aschempp) [reply tests/Routing/UrlGeneratorTest.php]: Fixed in 88d9d28703ef82958c464bb4909698b0eb206ced

**Comment by leofeyer** [comment]
Merged in 1a1e035ba0b44d167092e30131a3a33f75720579 (`hotfix/4.4.5`).
